### ISSUE TITLE
Wrongly activated dot on line chart with multiple data sources and numeric X axis

### ISSUE THREAD

**Issue by MarwanDebbiche** [recharts/recharts#3044]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-forked-cfmcl1?file=/package.json)

### Steps to reproduce
When making a line chart with multiple lines and a tooltip, each line with its own data source, and with a numeric X Axis, there is a bug with the activated dots and the tooltip.

### What is expected?
Only activates dots with the same X axis value.

### What is actually happening?
When hovering on a point from one line, sometimes a point from another line with a very different X axis value gets activated and appears in the tooltip.
(E.g. when hovering over the point with xAxis = 30 from the code sandbox reproduction, the point from the other line with xAxis = 13 gets activated as well, and appears in the tooltip)

| Environment | Info |
|---|---|
| Recharts | v2.1.6 |
| React | react@18.2.0 |
| System | MacOS |
| Browser | Chrome, Safari |

**Comment by leadq** [comment]
I debugged the source code and saw that activeDot is handled by chart and calculate index of *active* xAxis tick (think like column but I dont know how yet). Probably index of tick is calculated wrong in your case. I am also facing another issue with active dot and no solution I found. It is like, when you draw composed chart with mix of bars and a line, activeDot on the line appears when I mouse hover the bar on the same tick. I cant manage active dot from outside. I set activeDot property of <Line, but position of activeDot still managed by chart and sometimes put "customActiveDot" component wrong index.

**Comment by florianheller** [comment]
I had the same problem with Unix timestamps on the X-Axis. 
As per https://github.com/recharts/recharts/issues/2885, set `allowDuplicatedCategory={false}` on the X-Axis to solve the problem.

**Comment by leadq** [comment]
@florianheller I dont understand what that property do. In documentation, it only says this

```
allowDuplicatedCategory: Boolean
Allow the axis has duplicated categorys or not when the type of axis is "category".
```

so ? what happens if I give it false or true. What does duplicated categories means, what is category of axis?

I just dont understand. If you know and make me informed, I will be appreciated

**Comment by florianheller** [comment]
This property should only apply to categorical axes. Having a dataset like [{food: "Burger", price: 2},{food: "Burger", price: 4}] should result in 2 separate entries on the X-Axis if _allowDuplicatedCategory_ is set to _true_, but will ignore the second entry if set to _false_. (Play around with the SimpleBarChart example). 
I don't know why the property acts on numerical axes, since it doesn't make much sense and should be false by default, but it solves the issue.

**Comment by ckifer** [comment]
Labelling this as a bug for triage

**Comment by homerchen19** [comment]
Still facing the same issue of numeric X Axis in version `2.10.3` 😕

**Comment by Knive** [comment]
I faced the same issue and somehow `allowDuplicatedCategory={false}` did the trick ✅ even though my `XAxis` component is of `type="number"`

**Comment by apiec** [comment]
Yeah, I encountered the very same problem on version `2.15.0` and it was very hard to google for it :P Similar setup with having the `<XAxis type="number"/>` without the `allowDuplicatedCategory={false}` and values being unix timestamps.

I noticed that the dots would be marked as active only while the cursor was on the left of the dot. If the cursor was on the right they wouldn't be marked as active and the tooltip would show garbage values (I think the last value of ones that should be displayed)

**Comment by PavelVanecek** [comment]
This is still a problem in 3.0: https://codesandbox.io/p/sandbox/recharts-issue-template-forked-yhhnky?file=%2Fpackage.json
